pkg/oci/pusher/pusher.go

```diff
@@ -163,8 +167,8 @@ func (p *Pusher) Push(ctx context.Context, artifactType oci.ArtifactType,
 		}
 	}
 
-	if artifactType == oci.Rulesfile {
-		// We should have only one manifestDesc.
+	if artifactType != oci.Plugin {
```

Comment:
I'd make the check more explicit here (ie:
```suggestion
	if artifactType == oci.Rulesfile || artifactType == oci.Asset {
```

Reply:
True, it's better being explicit if we add another one too!